lib/src/document.dart

```diff
@@ -28,13 +38,31 @@ class Document {
     this.linkResolver,
     this.imageLinkResolver,
     this.encodeHtml = true,
-  }) : extensionSet = extensionSet ?? ExtensionSet.commonMark {
-    _blockSyntaxes
-      ..addAll(blockSyntaxes ?? [])
-      ..addAll(this.extensionSet.blockSyntaxes);
-    _inlineSyntaxes
-      ..addAll(inlineSyntaxes ?? [])
-      ..addAll(this.extensionSet.inlineSyntaxes);
+    this.withDefaultBlockSyntaxes = true,
+    this.withDefaultInlineSyntaxes = true,
+  }) {
+    hasCustomInlineSyntaxes = inlineSyntaxes?.isNotEmpty == true ||
```

Comment:
We should be able to set this in an initializer:

```dart
    this.withDefaultInlineSyntaxes = true,
  }) :
      hasCustomInlineSyntaxes = ...
  {
```

Then the field can be `final` without `late`.

Reply:
Fixed